airflow/io/path.py

```diff
@@ -40,21 +40,15 @@
 
 PT = typing.TypeVar("PT", bound="ObjectStoragePath")
 
-default = "file"
-
 
 class _AirflowCloudAccessor(_CloudAccessor):
     __slots__ = ("_store",)
 
-    def __init__(self, parsed_url: SplitResult | None, **kwargs: typing.Any) -> None:
-        store = kwargs.pop("store", None)
-        conn_id = kwargs.pop("conn_id", None)
-        if store:
-            self._store = store
-        elif parsed_url and parsed_url.scheme:
+    def __init__(self, parsed_url: SplitResult | None, conn_id: str, **kwargs: typing.Any) -> None:
```

Comment:
This overrides the __init__ constructor of the parent (CloudAccessor). While we are not using it in that way, I wanted to keep it the same so it is interchangeable.

Reply:
Would `conn_id: str | None = None` make this more comfortable? The interface difference is not a tachnically relevant issue IMO, but making the argument optional is not a big deal to me.